sway-lib-std/src/u256.sw

```diff
@@ -88,4 +107,268 @@ impl U256 {
     pub fn bits() -> u32 {
         256
     }
+
+    /// Get 4 64 bit words from a single U256 value.
+    fn decompose(self) -> (u64, u64, u64, u64) {
+        (self.a, self.b, self.c, self.d)
+    }
+}
+
+impl core::ops::Ord for U256 {
+    pub fn gt(self, other: Self) -> bool {
+        self.a > other.a || self.a == other.a && self.b > other.b ||  self.a == other.a && self.b == other.b  && self.c > other.c ||  self.a == other.a && self.b == other.b  && self.c == other.c && self.d > other.d
+    }
+
+    pub fn lt(self, other: Self) -> bool {
+        self.a < other.a || self.a == other.a && self.b < other.b ||  self.a == other.a && self.b == other.b  && self.c < other.c ||  self.a == other.a && self.b == other.b  && self.c == other.c && self.d < other.d
+    }
+}
+
+impl core::ops::BitwiseAnd for U256 {
+    pub fn binary_and(val: self, other: Self) -> Self {
+        let(value_word_1, value_word_2, value_word_3, value_word_4) = val.decompose();
+        let(other_word_1, other_word_2, other_word_3, other_word_4) = other.decompose();
+        let word_1 = value_word_1 & other_word_1;
+        let word_2 = value_word_2 & other_word_2;
+        let word_3 = value_word_3 & other_word_3;
+        let word_4 = value_word_4 & other_word_4;
+        let rebuilt = ~U256::from(word_1, word_2, word_3, word_4);
+        rebuilt
+    }
+}
+
+impl core::ops::BitwiseOr for U256 {
+    pub fn binary_or(val: self, other: Self) -> Self {
+        let(value_word_1, value_word_2, value_word_3, value_word_4) = val.decompose();
+        let(other_word_1, other_word_2, other_word_3, other_word_4) = other.decompose();
+        let word_1 = value_word_1 | other_word_1;
+        let word_2 = value_word_2 | other_word_2;
+        let word_3 = value_word_3 | other_word_3;
+        let word_4 = value_word_4 | other_word_4;
+        let rebuilt = ~U256::from(word_1, word_2, word_3, word_4);
+        rebuilt
+    }
+}
+
+impl core::ops::BitwiseXor for U256 {
+    pub fn binary_xor(val: self, other: Self) -> Self {
+        let(value_word_1, value_word_2, value_word_3, value_word_4) = val.decompose();
+        let(other_word_1, other_word_2, other_word_3, other_word_4) = other.decompose();
+        let word_1 = value_word_1 ^ other_word_1;
+        let word_2 = value_word_2 ^ other_word_2;
+        let word_3 = value_word_3 ^ other_word_3;
+        let word_4 = value_word_4 ^ other_word_4;
+        let rebuilt = ~U256::from(word_1, word_2, word_3, word_4);
+        rebuilt
+    }
+}
+
+impl core::ops::Shiftable for U256 {
+    fn lsh(self, shift_amount: u64) -> Self {
```

Comment:
I didn't realize that we would be able to use operators like `<<` if a trait was implemented in `std`. I thought it had to be implemented in `core` for that to work, but I guess because the actual traits are defined in core this works.